python/langsmith/run_trees.py

```diff
@@ -210,6 +211,10 @@ def set(
                 self.outputs = {}
             else:
                 self.outputs = dict(outputs)
+        if usage is not NOT_PROVIDED:
+            self.extra.setdefault("metadata", {}).setdefault(
+                "usage_metadata", {}
+            ).update(usage or {})
```

Comment:
nit, should we set this instead of updating? think id expect a `set` method to fully override any existing values

Reply:
You are probably right yeah